### DISCUSSION TITLE
VB scripting and interactive revisited

### DISCUSSION THREAD

**Discussion by govert** [dotnet/roslyn#50584] [General]
Some years ago, work on supporting VB in the Roslyn scripting libraries and the VS interactive console was paused, and I understand never released.
In the context of revived VB support in .NET 5+, I'd like to ask what the thoughts are around improvements and a release of the VB scripting and interactive support in Roslyn. The higher-level consumers of this feature like the 'dotnet script' runner and the 'dotnet/interactive' kernel can be poked at quite independently, mostly as a fork of the C# versions. But for the core language part, I believe some support inside the Roslyn compiler would help a lot. So:

* Would discussion and contributions to revisit VB scripting in Roslyn be considered?
* Issue https://github.com/dotnet/roslyn/issues/13523 seems to represent the last discussion on the topic.
* Issue https://github.com/dotnet/roslyn/issues/6554 seems to be the main tracking item from 2015. Would that still be a good starting point?
* This give a list of the open issues, mostly indicating "help wanted": https://github.com/dotnet/roslyn/issues?utf8=%E2%9C%93&q=is%3Aopen%20label%3AInteractive-ScriptingLogic%20label%3ALanguage-VB

**Comment by paul1956** [comment]
@govert, @dualbrain, @VBAndCs The design has changed significantly from 2015. To create the missing scripting package (requested separately) looks trivial it is just a matter of adding a project reference to the C# package but that doesn't give you VB Interactive. There is a whole project that needs to support VB and then some way to enable Visual Studio to open the Window. The simple way would be to clone the C# project and change it to support VB but that is not how "Roslyn does things". In the Roslyn way the current interactive project would need to be refactored into common, VB and C# components where generally the common components are written in C#. I believe the first step is to do the refactoring and get the C# part working again, then add the VB stuff.

**Reply by paul1956** [reply]
@govert Please comment on #51062

**Reply by govert** [reply]
Thank you @paul1956 - I've added a bit more of a story there https://github.com/dotnet/roslyn/issues/51062#issuecomment-775749582

To me the Roslyn plumbing precedes and is much more important than an interactive VB window in VS. If fact, the latter would probably be easier as a separate extension at first.

**Reply by paul1956** [reply]
The plumbing package looks trivial, you can just add the VB scripting project to C# NuGet package. I looks like a few line change. I did it and it builds no idea what to do with it from there so I am not going do do a PR.
